cve_bin_tool/data_sources/gad_source.py

```diff
@@ -189,14 +189,30 @@ def parse_range_string(self, range_string):
             versions = version_string.replace(",", " ").split(" ")
 
             for version in versions:
+                if (
+                    version.replace(")", "")
+                    .replace("(", "")
+                    .replace("[", "")
+                    .replace("]", "")
+                    == ""
+                ):
```

Comment:
Ugh, this works but it's kind of confusing to do actual string replaces here so it probably needs a comment or possibly needs to be written in a different way?  

I *think* what you're trying to do is make sure there's at least one character in here that isn't a bracket.  Since a regex with a bunch of brackets in it is just as hard to read as this is (`"[[\]()]"`), you could maybe go the other way and look for an actual version number to improve readability:

```suggestion
                # Make sure we have an actual version number and not just a bunch of brackets
                if not re.search("[0-9]", version):
```

This will fail if we have versions that are only letters, but I'm pretty sure those don't work right now for other reasons.

BTW, you probably can't just accept this suggestion without making changes because I'm not sure there's an `import re` line above. And you don't need to accept my suggestion if you just put an explanation in a comment so it's clear what you're doing.

Reply:
thanks @terriko 
this seems like the best way to do this.